src/applicationManager.ts

```diff
@@ -528,3 +546,239 @@ async function getGraphAPIRoles(token: string, graphApiPermissions: string[]) {
   }
   return { appRoleIds, graphAPIObjectId };
 }
+
+/**
+ * Updates an Azure Entra Application, with a collection of App Roles
+ * @param applicationId The object ID of application registration
+ * @param appRoles appRoles to update the application with
+ */
+export async function updateApplicationAppRoles({
+  token,
+  applicationId,
+  appRoles,
+}: {
+  token: string;
+  applicationId: string;
+  appRoles: AppRole[];
+}): Promise<void> {
+  let body = {
+    appRoles: appRoles,
+  };
+  const result = await fetch(
```

Comment:
I have a question : the ticket does not clarify which object  the app roles shall be defined to `application` object or `service principal`? seems `application` object is assumed. 

Background info: https://learn.microsoft.com/en-us/graph/api/resources/approle?view=graph-rest-1.0#json-representation

To add, update, or remove app roles for an application, [update the application](https://learn.microsoft.com/en-us/graph/api/application-update?view=graph-rest-1.0) for the app or service. App roles on the application entity will be available in all tenants where the application is used. To define app roles that are only applicable in your tenant (for example, app roles representing custom roles in your instance of a multitenant application), you can also [update the service principal](https://learn.microsoft.com/en-us/graph/api/serviceprincipal-update?view=graph-rest-1.0) for the app, to add or update app roles to the appRoles collection.